dot/rpc/modules/chain_test.go

```diff
@@ -2,3 +2,494 @@
+	}
+	type args struct {
+		r   *http.Request
+		req *ChainFinalizedHeadRequest
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		expErr error
+		exp    ChainHashResponse
+	}{
+		{
+			name: "GetFinalisedHash OK",
+			fields: fields{
+				mockBlockAPI,
+			},
+			args: args{
+				req: &ChainFinalizedHeadRequest{
+					Round: uint64(21),
+					SetID: uint64(21),
+				},
+			},
+			exp: expRes,
+		},
+		{
+			name: "GetFinalisedHash ERR",
+			fields: fields{
+				mockBlockAPIErr,
+			},
+			args: args{
+				req: &ChainFinalizedHeadRequest{
+					Round: uint64(21),
+					SetID: uint64(21),
+				},
+			},
+			expErr: errors.New("GetFinalisedHash Error"),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			cm := &ChainModule{
+				blockAPI: tt.fields.blockAPI,
+			}
+			res := ChainHashResponse(nil)
+			err := cm.GetFinalizedHeadByRound(tt.args.r, tt.args.req, &res)
+			if tt.expErr != nil {
+				assert.EqualError(t, err, tt.expErr.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+			assert.Equal(t, tt.exp, res)
+		})
+	}
+}
+
+func TestChainModule_GetHeader(t *testing.T) {
+	emptyHeader := types.NewEmptyHeader()
+	testHash := common.NewHash([]byte{0x01, 0x02})
+	inputHash, err := common.HexToHash("0x0102000000000000000000000000000000000000000000000000000000000000")
+	require.NoError(t, err)
+
+	mockBlockAPI := new(apimocks.BlockAPI)
+	mockBlockAPI.On("GetHeader", inputHash).Return(emptyHeader, nil)
+
+	mockBlockAPIErr := new(apimocks.BlockAPI)
+	mockBlockAPIErr.On("GetHeader", inputHash).Return(nil, errors.New("GetFinalisedHash Error"))
+
+	expRes, err := HeaderToJSON(*emptyHeader)
+	require.NoError(t, err)
+
+	type fields struct {
+		blockAPI BlockAPI
+	}
+	type args struct {
+		r   *http.Request
+		req *ChainHashRequest
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		expErr error
+		exp    ChainBlockHeaderResponse
+	}{
+		{
+			name: "GetHeader OK",
+			fields: fields{
+				mockBlockAPI,
+			},
+			args: args{
+				req: &ChainHashRequest{&testHash},
+			},
+			exp: expRes,
+		},
+		{
+			name: "GetHeader ERR",
+			fields: fields{
+				mockBlockAPIErr,
+			},
+			args: args{
+				req: &ChainHashRequest{&testHash},
+			},
+			expErr: errors.New("GetFinalisedHash Error"),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			cm := &ChainModule{
+				blockAPI: tt.fields.blockAPI,
+			}
+			res := ChainBlockHeaderResponse{}
+			err := cm.GetHeader(tt.args.r, tt.args.req, &res)
+			if tt.expErr != nil {
+				assert.EqualError(t, err, tt.expErr.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+			assert.Equal(t, tt.exp, res)
+		})
+	}
+}
+
+func TestChainModule_ErrSubscriptionTransport(t *testing.T) {
+	req := &EmptyRequest{}
+	res := &ChainBlockHeaderResponse{}
+	cm := NewChainModule(new(apimocks.BlockAPI))
+
+	err := cm.SubscribeFinalizedHeads(nil, req, res)
+	require.Equal(t, err, ErrSubscriptionTransport)
```

Comment:
nit Use `require.ErrorIs` instead? As in Equal works but ErrorIs sounds more appropriate here (in case we wrap the error later too)

Same for below calls.